index.js

```diff
@@ -64,82 +83,146 @@ function withFilter(predicate) {
 }
 exports.withFilter = withFilter;
 
-function createWriter(predicate) {
-  return transit.writer('json', {
-      handlers: transit.map([
-        Immutable.Map, transit.makeWriteHandler({
-          tag: function() {
-            return 'iM';
-          },
-          rep: function(m) {
-            var i = 0, a = new Array(2 * m.size);
-            if (predicate) {
-              m = m.filter(predicate);
-            }
-            m.forEach(function(v, k) {
-              a[i++] = k;
-              a[i++] = v;
-            });
-            return a;
-          }
-        }),
-        Immutable.OrderedMap, transit.makeWriteHandler({
-          tag: function() {
-            return 'iOM';
-          },
-          rep: function(m) {
-            var i = 0, a = new Array(2 * m.size);
-            if (predicate) {
-              m = m.filter(predicate);
-            }
-            m.forEach(function(v, k) {
-              a[i++] = k;
-              a[i++] = v;
-            });
-            return a;
-          }
-        }),
-        Immutable.List, transit.makeWriteHandler({
-          tag: function() {
-            return "iL";
-          },
-          rep: function(v) {
-            if (predicate) {
-              v = v.filter(predicate);
-            }
-            return v.toArray();
-          }
-        }),
-        Immutable.Set, transit.makeWriteHandler({
-          tag: function() {
-            return "iS";
-          },
-          rep: function(v) {
-            if (predicate) {
-              v = v.filter(predicate);
-            }
-            return v.toArray();
-          }
-        }),
-        Immutable.OrderedSet, transit.makeWriteHandler({
-          tag: function() {
-            return "iOS";
-          },
-          rep: function(v) {
-            if (predicate) {
-              v = v.filter(predicate);
-            }
-            return v.toArray();
-          }
-        }),
-        Function, transit.makeWriteHandler({
-          tag: function() {
-            return '_';
-          },
-          rep: function() {
-            return null;
-          }
-        })
-      ])
+function withRecords(records) {
+  var recordMap = {};
+
+  records.forEach(function(RecordType) {
+    var rec = new RecordType({});
+    var recName = recordName(rec);
+
+    if (!recName || recName === 'Record') {
+      throw new Error('Cannot (de)serialize Record() without a name');
+    }
+
+    recordMap[recName] = RecordType;
```

Comment:
I wonder if this will also run into problems with namespaces, as the record names don't include the namespace.

Perhaps this should also check that all names are unique and error if they aren't?